### ISSUE TITLE
connectRange does not trigger an update when both range bounds are set to undefined

### ISSUE THREAD

**Issue by JJWesterkamp** [algolia/instantsearch.js#4626]
## 🐛 Bug description

I'm working on a `connectRange()` implementation with a custom slider library (noUiSlider) and have run into the issue that `connectRange` will not trigger an update when I call `refine([undefined, undefined])`. I have tracked down the code that ultimately causes this behaviour, which is [this `if`-statement][issue-source] in the `connectRange` source code.

I have also found that this issue **only occurs** when I have an override for the `onStateChange` method on the `InstantSearchOptions` object passed to the `instantsearch()` factory function. If I remove this override everything works as expected.
 
## 🔍 Bug reproduction

1. Create and register a `connnectRange()` widget
2. Have an override of `onStateChange()` in the configuration passed to the `instantsearch()` factory function, exactly as [the documentation][onstatechange-doc] shows:

    ```typescript
    {
        onStateChange({ uiState, setUiState })
        {
            setUiState(uiState)
        }
    }
    ```
3. Restrict one bound of the range (UI updates as expected)
4. Reset that bound to match the corresponding range min/max value. (UI does not update)

**Live reproduction:**

In this [live example][live-example] perform the following steps:
1. click down the max-value of the price number input, and watch the results count go down
2. click back up the max-value, then the results count should go back up, but it doesn't. Nor does the hits list update as expected.
3. click on any pagination link, and watch the results count update now. 

    _Because of the observation in step 3, it looks like it that the bounds are updated internally, because if I have any other interaction afterwards that triggers an update of the search results, the reset bounds do get noticed and my UI updates using the `undefined` bounds._

4. Comment (disable) the `onStateChange` override
5. Perform steps 1 to 3 again, and see everything work as expected.

## 💭 Expected behavior

The [refine() documentation][refine-docblock] states:

> You can use `undefined` to remove a previously set bound or to set an infinite bound.

This makes me expect that when the state is for instance `[10, undefined]` and then I call `refine([undefined, undefined])` that the lower bound `10` is being removed, and then a new search is triggered, regardless of other non-related options such as the `onStateChange` method.

## Environment

- OS: macOS
- Browser: Chrome - Version 87.0.4280.88 (Official Build) (x86_64)
- instantsearch.js version: `4.10.0-experimental-typescript.0`

[live-example]: https://codesandbox.io/s/agitated-bhaskara-s78pq
[onstatechange-doc]: https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/#widget-param-onstatechange
[refine-docblock]: https://github.com/algolia/instantsearch.js/blob/v4.11.0/src/connectors/range/connectRange.ts#L38
[issue-source]: https://github.com/algolia/instantsearch.js/blob/v4.11.0/src/connectors/range/connectRange.ts#L329

**Comment by eunjae-lee** [comment]
Thank you @JJWesterkamp a lot for your effort to open this issue.
We'll get it fixed and you can follow the progress via #4627.

**Comment by JJWesterkamp** [comment]
@eunjae-lee Thank you very much!

**Comment by JJWesterkamp** [comment]
@eunjae-lee I'm not sure if it is related to this issue, but I also found that the initial `renderOptions` passed to the connector callback argument are incorrect if a range filter is set at initialisation time. For example:

```typescript
connectRange<CustomPriceRangeOptions>((renderOptions, isFirstRender) => {
    renderOptions.start[0] // Should be the initial lower bound filter if set
    renderOptions.start[1] // Should be the initial upper bound filter if set
})
```

But regardless of how I supply this state, I always get passed the range limits min/max. I have tried:
1. Setting the `initialUiState`:
    ```typescript
    instantsearch({
        // ...
        initialUiState: {
            [indexName]: {
                range: {
                    price: '150:1500',
                }
            },
        },
    })
    ```
2. Use state mapping with the history router:
    ```typescript
    routeToState(routeState: RouteState): UiState {
        return {
            [indexName]: {
                range: {
                    price: '150:1500'
                },
            },
        }
    }
    ```

Am I missing something here?

Thanks in advance!

**Comment by eunjae-lee** [comment]
Hey @JJWesterkamp I just tested it and in https://github.com/algolia/instantsearch.js/pull/4627 it will be fixed as well.
I just added a test to make sure it works there :)

**Comment by eunjae-lee** [comment]
Hello @JJWesterkamp , this has been released in 4.13.2
